x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+outputs:
```

Comment:
What do `default`, `long_term_storage` and `monitoring` mean? Am I correct in assuming that when writing data to Elasticsearch that each datatype Beats/Agent/Endpoint write will be hardcoded to go to particular destination? Is there room for more outputs if Endpoint ends up needing another Endpoint-specific output? Is there any way to control what index is written to for an output?

Reply:
I've discussed with @ferullo on zoom about this. but for clarity I will stil add part of the discussion here. The `default`, `long_term_storage` and `monitoring` are destination, in the initial version we would only support `default` and `long_term_storage`.   When you define a datasource you can use the **use_output** option to defines to which named output to send the data.

So yes, if the endpoint need a special output destination it could be added.

Concerning the **index**, the index is not something that a user would be able to configure directly we do this to hide the complexity inherent to manage indices (templates mapping, ilm policies and dashboard) instead the agent is using **the type of the input**, the **dataset** and the **defined namespace** to generate the appropriate index destination. The only user editable part is the namespace, which is used as a grouping mechanism like "production" data.

I presume that endpoint should try to align with that indexing strategy and allow the user to set up the namespace maybe its something to discuss with @ruflin